docs/decisions/handle-nodata-in-rastercubes.md

```diff
@@ -0,0 +1,69 @@
+# How to handle nodata values in Rastercubes
+
+## Context
+The logic processes as defined by OpenEO can return True/False/null. In numpy there is no sentinel value for missing data in boolean/integer/char arrays (like there is with NaN for float).
+
+See [this blog](https://jakevdp.github.io/PythonDataScienceHandbook/03.04-missing-values.html) for a good overview to missing data handling in Python. Also see [NEP-26](https://numpy.org/neps/nep-0026-missing-data-summary.html) for the state of this issue in the python community.
+
+To implement `and` as specified, instead of this:
+
+**Example A: What you'd expect from an implementation for a logical `and` process**
+
+logical_and = np.logical_and(x, y)
+
+we have to do this:
+
+**Example B: Workaround we have to use to account for null values are returned by logic processes**
+
+nan_x = np.isnan(x)
+nan_y = np.isnan(y)
+xy = np.logical_and(x, y)
+nan_mask = np.logical_and(nan_x, xy)
+xy = np.where(~nan_mask, xy, np.nan)
+nan_mask = np.logical_and(nan_y, xy)
+xy = np.where(~nan_mask, xy, np.nan)
+return xy
+
+Note that the output array of Example A has the dtype `bool_`, whereas the output array of Example B has dtype `float64`. This is because the missing data value `np.nan` is only defined for dtype `float64` and thus the entire array needs to be upcast to `float64`.
+
+See these numpy snippets to confirm this:
+
+>>> np.array([True, False])
+array([ True, False], dtype='bool')  # boolean array
+>>> np.array([True, False]).itemsize
+1  # one byte per item
+>>> np.array([True, False, np.nan])
+array([ 1., 0., nan], dtype='float64')  # casts to float64
+>>> np.array([True, False, np.nan]).itemsize
+8
```

Comment:
```suggestion
```python
>>> np.array([True, False])
array([ True, False], dtype='bool')  # boolean array
>>> np.array([True, False]).itemsize
1  # one byte per item
>>> np.array([True, False, np.nan])
array([ 1., 0., nan], dtype='float64')  # casts to float64
>>> np.array([True, False, np.nan]).itemsize
8
```
```